File: Server/Transitions.py
from Engine import EngineCard
from Planet import Planet
from Util import Color


def transition(player: Color, blue: Planet, red: Planet, engineCard: EngineCard):
    if engineCard == EngineCard.SWAP or engineCard == EngineCard.PROBE:
        raise Exception("Handle SWAP and PROBE on your own! They aren't worth my time")

    # For now, the comment "CNOT: flip your ship only if the other ship is on CENTARIOUS ONE" was ignored
    if player == Color.Blue:
        new = blue.get_transition(engineCard, player)
        if isinstance(new, Planet):
            return new
        else:
            return new[red]
    else:
        new = red.get_transition(engineCard, player)
        if isinstance(new, Planet):
            return new
        else:
            return new[blue]

File: Server/JavaScript/Transitions.js
function transition(player_color, blue_planet, red_planet, engine_card) {
    if (engine_card === EngineCard.SWAP || engine_card === EngineCard.PROBE) {
        throw new Error("Handle SWAP and PROBE on your own; they aren't worth my time and energy !");
    }

    // For now, the comment "CNOT: flip your ship only if the other ship is on CENTARIOUS ONE" was ignored
    if (player_color === Color.Blue) {
        var destination_function = blue.get_transition(engine_card, player_color);
        return destination_function(red_planet);
    } else {
        var destination_function = red.get_transition(engine_card, player_color);
        return destination_function(blue_planet);
    }
}